File: comparisons/parity/glmm_probit.jl
# Probit GLMM parity exporter (MixedModels.jl).
#   julia comparisons/parity/glmm_probit.jl

using CSV
using DataFrames
using JSON
using MixedModels

file_path = joinpath("tests", "data", "cbpp_binary.csv")
isfile(file_path) || error("missing dataset: $file_path")

df = CSV.read(file_path, DataFrame)
df.herd = string.(df.herd)

form = @formula(y ~ 1 + period2 + period3 + period4 + (1 | herd))
m = fit(MixedModel, form, df, Binomial(), ProbitLink())

coef_names = coefnames(m)
coef_vals = coef(m)
coefficients = Dict(zip(coef_names, coef_vals))

payload = Dict(
    "case" => "glmm_probit",
    "implementation" => "julia",
    "coefficients" => coefficients,
)
println(JSON.json(payload))

File: comparisons/parity/glmm_probit.R
source("comparisons/parity/parity_r_setup.R")
suppressPackageStartupMessages({
  library(lme4)
  library(jsonlite)
})
source("comparisons/r_setup.R")

data <- read.csv("tests/data/cbpp_binary.csv")
data$herd <- as.factor(data$herd)
fit <- glmer(
  y ~ period2 + period3 + period4 + (1 | herd),
  data = data,
  family = binomial(link = "probit")
)
coefs <- as.list(as.numeric(fixef(fit)))
names(coefs) <- names(fixef(fit))
cat(toJSON(
  list(case = "glmm_probit", implementation = "r", coefficients = coefs),
  auto_unbox = TRUE,
  digits = 10
))
